lib/utils/index.js

```diff
@@ -520,5 +520,56 @@ module.exports = {
         }
       }
     }
+  },
+
+  /**
+   * Find all function witch do not allways return values
+   * @param {Array} nodes an array nodes
+   * @param {boolean} treatUndefinedAsUnspecified
+   */
+  findMissingReturns (nodes, treatUndefinedAsUnspecified) {
```

Comment:
You should avoid mutating arguments. I suggest to modify this function to be more explicit (we can even change the name a bit, to be consistent with `executeOnVue` and so on):
```
executeOnFunctionsWithoutReturn (treatUndefinedAsUnspecified, cb) {
  // ...

  return {
    // ...
    'ArrowFunctionExpression:exit' (node) {
      if (isValidReturn()) return
      cb(funcInfo.node)
    },
    'FunctionExpression:exit' (node) {
      if (isValidReturn()) return
      cb(funcInfo.node)
    }
  }
}
```

Then you could use it like this:
```
return Object.assign({},
  utils.executeOnFunctionsWithoutReturn(treatUndefinedAsUnspecified, node => {
    forbiddenNodes.push(node)
  }),
  utils.executeOnVue(context, properties => {
    // ...
  })
)
```

I think it will be much easier to understand, and what do you think @armano2 ?